apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClientImpl.SingleChannel.cs

```diff
@@ -276,10 +276,13 @@ private void StopStreamingPull()
                 // Ignore all errors; the stream may be in any state.
                 try
                 {
-                    _registerTaskFn(_pull.WriteCompleteAsync());
+                    Add(_pull.WriteCompleteAsync(), Next(false, _pull.Dispose));
```

Comment:
Does the `false` mean that the previous task does not need to have succeede? Checking as we'd need Dispose to execute regardless of whether WriteCompleteAsync succeeded or not.

Reply:
Ok my main question is if Dispose will execute even if WriteCompleteAsync fails? That's now looping the loop but just to make sure we really dispose whenever we can. If not, maybe register a single task that calls WriteCompleteAsync in a try and Dispose in the finally?

Reply:
Thanks Amanda. To be honest, I wasn't 100% sure. But, I have updated second commit and now I am 100% sure that Dispose will be called even if WriteCompleteAsync fails. PTAL. Thanks.